File: 7KYU/Recursion 101/solution.clj
(ns kata.core)

(defn solve
  [a b]
  (cond (or (zero? a) (zero? b)) (list a b)
        (>= a (* 2 b)) (recur (- a (* 2 b)) b)
        (>= b (* 2 a)) (recur a (- b (* 2 a)))
        :else (list a b)))

File: 7KYU/Recursion 101/solution.go
package kata

func Solve(a, b int) []int {
	if a == 0 || b == 0 {
		return []int{a, b}
	} else if a >= 2*b {
		return Solve(a-2*b, b)
	} else if b >= 2*a {
		return Solve(a, b-2*a)
	} else {
		return []int{a, b}
	}
}
